File: api/src/main/java/net/blueshell/api/listener/email/EventSignupEmailEventListener.java
package net.blueshell.api.listener.email;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import net.blueshell.api.common.event.job.EventSignupEmailEvent;
import net.blueshell.api.job.email.EventSignupEmailJob;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class EventSignupEmailEventListener {

    private final EventSignupEmailJob job;

    @EventListener
    public void onSend(EventSignupEmailEvent evt) {
        Long id = evt.eventSignUpId();
        if (id == null) return;
        job.send(id);
    }
}

File: api/src/main/kotlin/net/blueshell/api/listener/email/EventSignupEmailEventListener.kt
package net.blueshell.api.listener.email

import net.blueshell.api.common.event.job.EventSignupEmailEvent
import net.blueshell.api.job.email.EventSignupEmailJob
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class EventSignupEmailEventListener(
    private val job: EventSignupEmailJob
) {
    @EventListener
    fun onSend(evt: EventSignupEmailEvent) {
        val id = evt.eventSignUpId
        if (id == null) return
        job.send(id)
    }
}
